### PR TITLE
Allow multiple other dimensions in prewhiten data

### PR DESCRIPTION
**Pull request by fzimmermann89**
and use triangular solve instead of torch.inv as it is supposedly more numerically stable.

### CODE REVIEW

**Comment by fzimmermann89** [comment]
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #270
* __->__ #272
* #269
* #268
* #267

File: tests/algorithms/test_prewhiten_kspace.py [hunk 1]
```diff
@@ -32,19 +33,23 @@ def _test_prewhiten_kspace(random_kheader, device):
 
     # Dimensions
     n_coils = 4
-    n_kx = 128
+    ns_k2k1k0 = (4, 5, 32)
```
Comment (ckolbPTB) [inline tests/algorithms/test_prewhiten_kspace.py]: n_k2k1k0 and n_other?

File: tests/algorithms/test_prewhiten_kspace.py [hunk 2]
```diff
@@ -32,19 +33,23 @@ def _test_prewhiten_kspace(random_kheader, device):
...
+    ns_other = (1,)
```
Comment (ckolbPTB) [inline tests/algorithms/test_prewhiten_kspace.py]: would make sense to test for n_other > 1 if this is the feature this PR adds

**Review by ckolbPTB — CHANGES_REQUESTED** [review]

File: tests/algorithms/test_prewhiten_kspace.py (hunk 2 above)
Comment (fzimmermann89) [reply tests/algorithms/test_prewhiten_kspace.py]: 🙈

**Review by ckolbPTB — APPROVED** [review]

**Pull request merged**